src/code/z_play.c

```diff
@@ -540,13 +2060,280 @@ void Play_AssignPlayerActorCsIdsFromScene(GameState* thisx, s32 startActorCsId)
+        Camera_ChangeStatus(&this->subCameras[i], CAM_STATUS_INACTIVE);
+    }
+
+    this->cameraPtrs[CAM_ID_MAIN] = &this->mainCamera;
+    this->cameraPtrs[CAM_ID_MAIN]->uid = CAM_ID_MAIN;
+    this->activeCamId = CAM_ID_MAIN;
+
+    func_800DFF18(&this->mainCamera, 0x7F);
+    Sram_Alloc(&this->state, &this->sramCtx);
+    Construct_InitRegs(this);
+    Message_Init(this);
+    GameOver_Init(this);
+    SoundSource_InitAll(this);
+    EffFootmark_Init(this);
+    Effect_Init(this);
+    EffectSS_Init(this, 100);
+    CollisionCheck_InitContext(this, &this->colChkCtx);
+    AnimationContext_Reset(&this->animationCtx);
+    Cutscene_Init(this, &this->csCtx);
+
+    if (gSaveContext.nextCutsceneIndex != 0xFFEF) {
+        gSaveContext.save.cutscene = gSaveContext.nextCutsceneIndex;
+        gSaveContext.nextCutsceneIndex = 0xFFEF;
+    }
+
+    if (gSaveContext.save.cutscene == 0xFFFD) {
+        gSaveContext.save.cutscene = 0;
+    }
+
+    if (gSaveContext.nextDayTime != 0xFFFF) {
+        gSaveContext.save.time = gSaveContext.nextDayTime;
+        gSaveContext.skyboxTime = gSaveContext.nextDayTime;
+    }
+
+    if ((gSaveContext.save.time >= CLOCK_TIME(18, 0)) || (gSaveContext.save.time < CLOCK_TIME(6, 30))) {
+        gSaveContext.save.isNight = true;
+    } else {
+        gSaveContext.save.isNight = false;
+    }
+
+    func_800EDDB0(this);
+
+    if (((gSaveContext.gameMode != 0) && (gSaveContext.gameMode != 1)) || (gSaveContext.save.cutscene >= 0xFFF0)) {
+        gSaveContext.unk_3DC0 = 0;
+        Magic_Reset(this);
+        gSaveContext.sceneLayer = (gSaveContext.save.cutscene & 0xF) + 1;
+        gSaveContext.save.cutscene = 0;
+    } else {
+        gSaveContext.sceneLayer = 0;
+    }
+
+    sceneLayer = gSaveContext.sceneLayer;
+
+    Play_SpawnScene(
+        this, Entrance_GetSceneIdAbsolute(((void)0, gSaveContext.save.entrance) + ((void)0, gSaveContext.sceneLayer)),
+        Entrance_GetSpawnNum(((void)0, gSaveContext.save.entrance) + ((void)0, gSaveContext.sceneLayer)));
+    KaleidoScopeCall_Init(this);
+    func_80121FC4(this);
+
+    if (gSaveContext.nextDayTime != 0xFFFF) {
+        if (gSaveContext.nextDayTime == 0x8000) {
+            gSaveContext.save.day++;
+            gSaveContext.save.daysElapsed++;
+            gSaveContext.dogIsLost = true;
+            gSaveContext.nextDayTime = -2;
+        } else {
+            gSaveContext.nextDayTime = -3;
+        }
+    }
+
+    Play_InitMotionBlur();
+
+    R_PAUSE_MENU_MODE = 0;
+    R_PICTOGRAPH_PHOTO_STATE = 0;
+
+    PreRender_Init(&this->pauseBgPreRender);
+    PreRender_SetValuesSave(&this->pauseBgPreRender, D_801FBBCC, D_801FBBCE, NULL, NULL, NULL);
+    PreRender_SetValues(&this->pauseBgPreRender, D_801FBBCC, D_801FBBCE, NULL, NULL);
+
+    this->unk_18E64 = D_801FBB90;
+    this->pictoPhotoI8 = gPictoPhotoI8;
+    this->unk_18E68 = D_80784600;
+    this->unk_18E58 = D_80784600;
+    this->unk_18E60 = D_80784600;
```

Comment:
I'd like another pair of eyes on this, but these all seem to be various framebuffer pointers. Mainly I'm curious if `pictoPhotoI8` should have `fb` in its name somehow, or if it should even be a `TextuerPtr` or if its treated more as an image. I don't fully understand framebuffer stuff

Reply:
AFAIK it is not a framebuffer as it isn't storing frame data in the general sense, so doesn't need `fb`. I could see using TexturePtr, but it also is a known texture type so I think `u8*` works as well.

Reply:
I think best to wait on this one until we know we have all of the information.